app/imports/ui/components/NavBar.jsx

```diff
@@ -3,26 +3,36 @@ import PropTypes from 'prop-types';
 import { Meteor } from 'meteor/meteor';
 import { withTracker } from 'meteor/react-meteor-data';
 import { withRouter, NavLink } from 'react-router-dom';
-import { Menu, Dropdown, Header } from 'semantic-ui-react';
+import { Menu, Dropdown, Header, Loader, Icon } from 'semantic-ui-react';
 import { Roles } from 'meteor/alanning:roles';
+import { Profiles } from '/imports/api/profile/profile';
 
 /** The NavBar appears at the top of every page. Rendered by the App Layout component. */
 class NavBar extends React.Component {
   render() {
-    const menuStyle = { marginBottom: '10px' };
+    return (this.props.ready) ? this.renderNavBar() : <Loader active>Retrieving data</Loader>;
+  }
+  renderNavBar() {
+    const menuStyle = { marginBottom: '100px' };
+    //WHY IS _id UNDEFINED????????
+    // const { _id } = this.props.profile[0];
+
     return (
-      <Menu style={menuStyle} attached="top" borderless inverted>
+      <Menu style={menuStyle} borderless inverted fixed="top">
         <Menu.Item as={NavLink} activeClassName="" exact to="/">
           <Header inverted as='h1'>TeleSkope</Header>
         </Menu.Item>
-        {this.props.currentUser ? (
-            [<Menu.Item as={NavLink}
+        {Roles.userIsInRole(Meteor.userId(), 'student') || Roles.userIsInRole(Meteor.userId(), 'company') ? (
```

Comment:
extract Id first for cleaner code

Reply:
Did you mean assign Meteor.userId() to a variable? Here's what I did:
```
const userId = Meteor.userId();
...
{(Roles.userIsInRole(userId, 'student') || Roles.userIsInRole(userId, 'company')) ? (
```